src/include/simeng/arch/riscv/Instruction.hh

```diff
@@ -158,19 +164,25 @@ class Instruction : public simeng::Instruction {
   /** Retrieve the instruction's metadata. */
   const InstructionMetadata& getMetadata() const;
 
+  /** Retrieve the instruction's associated architecture. */
+  const Architecture& getArchitecture() const;
+
   /** A special register value representing the zero register. If passed to
    * `setSourceRegisters`/`setDestinationRegisters`, the value will be
    * automatically supplied as zero. */
   static const Register ZERO_REGISTER;
 
- private:
   /** The maximum number of source registers any supported RISC-V instruction
    * can have. */
-  static const uint8_t MAX_SOURCE_REGISTERS = 2;
+  static const uint8_t MAX_SOURCE_REGISTERS = 3;
   /** The maximum number of destination registers any supported RISC-V
    * instruction can have. */
   static const uint8_t MAX_DESTINATION_REGISTERS = 1;
 
+  /* Return the array of operand values */
+  const std::array<RegisterValue, MAX_SOURCE_REGISTERS> getOperands() const;
```

Comment:
Unless I'm missing something this function is un-implemented? And also seems redundant

Reply:
It is implemented at the bottom of ```instruction_execute.cc``` which is not the correct place for it so moved to ```instruction.cc```. It is used in he exception handler by the new functionality line 653